src/Sets/SparsePolynomialZonotope.jl

```diff
@@ -0,0 +1,296 @@
+export SparsePolynomialZonotope, expmat, nparams, ndependentgens, nindependentgens, ngens,
+        dependent_genmat, independent_genmat, indexvector,
+       linear_map, quadratic_map, remove_redundant_generators
+
+"""
+    SparsePolynomialZonotope{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, MNI<:AbstractVecOrMat{N}, ME<:AbstractMatrix{<:Integer}, VI<:AbstractVector{<:Integer}} <: AbstractPolynomialZonotope{N}
+
+Type that represents a sparse polynomial zonotope.
+
+A sparse polynomial zonotope ``\\mathcal{PZ} ⊂ \\mathbb{R}^n`` is represented by the set
+```math
+\\mathcal{PZ} = \\left\\{x \\in \\mathbb{R}^n : x = c + ∑ᵢ₌₁ʰ\\left(∏ₖ₌₁ᵖ α_k^{E_{k, i}} \\right)Gᵢ+∑ⱼ₌₁^qβⱼGIⱼ,~~ α_k ∈ [-1, 1]~~ ∀ i = 1,…,p, j=1,…,q \\right\\},
+```
+where ``c ∈ \\mathbb{R}^n`` is the offset vector (or center), ``G ∈ \\mathbb{R}^{n \\times h}`` is the dependent generator matrix with columns ``Gᵢ``,
+``GI ∈ \\mathbb{R}^{n×q}`` is the independent generator matrix and where ``E ∈ \\mathbb{N}^{p×h}_{≥0}`` is the exponent matrix with matrix elements ``E_{k, i}``.
+
+### Fields
+
+- `c` -- offset vector
+- `G` -- dependent generator matrix
+- `GI` -- independent generator matrix
+- `E` -- exponent matrix
+- `idx` -- identifiers vector
+
+### Notes
+
+Sparse polynomial zonotopes were introduced in [KA21].
+
+- [KA21] N. Kochdumper and M. Althoff. *Sparse Polynomial Zonotopes: A Novel Set Representation for Reachability Analysis*. Transactions on Automatic Control, 2021.
+"""
+struct SparsePolynomialZonotope{N,
+                                VN<:AbstractVector{N},
+                                MN<:AbstractMatrix{N},
+                                MNI<:AbstractVecOrMat{N},
+                                ME<:AbstractMatrix{<:Integer},
+                                VI<:AbstractVector{<:Integer}} <: AbstractPolynomialZonotope{N}
+    c::VN
+    G::MN
+    GI::MNI
+    E::ME
+    idx::VI
+
+    function SparsePolynomialZonotope(c::VN, G::MN, GI::MNI, E::ME, idx::VI) where {N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, MNI<:AbstractVecOrMat{N}, ME<:AbstractMatrix{<:Integer}, VI<:AbstractVector{<:Integer}}
+        @assert length(c) == size(G, 1) throw(DimensionMismatch("c and G should have the same number of rows"))
+        @assert length(c) == size(GI, 1) throw(DimensionMismatch("c and GI should have the same number of rows"))
+        @assert size(G, 2) == size(E, 2) throw(DimensionMismatch("G and E should have the same number of columns"))
+        @assert all(>=(0), E) throw(ArgumentError("E should have non-negative integers"))
+        @assert all(>(0), idx) throw(ArgumentError("identifiers in index vector must be positive integers"))
+        return new{N, VN, MN, MNI, ME, VI}(c, G, GI, E, idx)
+    end
+end
+
+function SparsePolynomialZonotope(c::AbstractVector, G::AbstractMatrix, GI::AbstractVecOrMat, E::AbstractMatrix{<:Integer})
+    n = size(E, 1)
+    idx = collect(1:n)
+    return SparsePolynomialZonotope(c, G, GI, E, idx)
+end
```

Comment:
```julia
julia> 1:4 isa AbstractVector{<:Integer}
true
```
Is there a reason to not use `1:n` directly?

Reply:
not really, updated